### DISCUSSION TITLE
Controller not registering through a Lora repeater

### DISCUSSION THREAD

**Discussion by veronal** [timmbogner/Farm-Data-Relay-System#210] [General]
Hello all,
I have a setup where I have a controller on the other side of a Lora repeater.
The only way I can get this to work is to first connect the controller to the main gateway (ID :1) let it register, then re-program the controller to talk to the Lora repeater (ID:2).  After I do this is works fine.  If the main gateway needs to reboot I must repeat the above to get the controller to work again.
It looks like the controller is not registering through the Lora repeater ?
Thanks in advance..

**Comment by timmbogner** [comment]
The intended process is for the controller to register with the repeater gateway. On the main gateway (UNIT_MAC 0x01) you should have it set to `#define LORA_NEIGHBOR_2 0x02`. It'll be helpful if you can post the following portions of both gateways' configs:


EXAMPLE_GATEWAY
```
//Addresses
#define UNIT_MAC           0x01  // The address of this gateway

#define ESPNOW_NEIGHBOR_1  0x00  // Address of ESP-NOW neighbor #1
#define ESPNOW_NEIGHBOR_2  0x00  // Address of ESP-NOW neighbor #2
#define LORA_NEIGHBOR_1    0x00  // Address of LoRa neighbor #1
#define LORA_NEIGHBOR_2    0x02  // Address of LoRa neighbor #2

// Interfaces
#define USE_ESPNOW  
//#define USE_LORA
//#define USE_WIFI  // Will cause errors if used with ESP-NOW. Use a serial link instead!
//#define USE_ETHERNET

// Routing
// Options: sendESPNowNbr(1 or 2); sendESPNowPeers(); sendLoRaNbr(1 or 2); broadcastLoRa(); sendSerial(); sendMQTT();
#define ESPNOWG_ACT    sendSerial();
#define LORAG_ACT      sendSerial();
#define SERIAL_ACT     sendESPNowNbr(2); sendESPNowPeers(); sendLoRaNbr(2); broadcastLoRa(); 
#define MQTT_ACT          
#define INTERNAL_ACT   sendSerial();
#define ESPNOW1_ACT    
#define ESPNOW2_ACT    sendSerial();                
#define LORA1_ACT      
#define LORA2_ACT      sendSerial();
```

**Reply by veronal** [reply]
Here the Gateway config (it looks to be correct)
```
//  FARM DATA RELAY SYSTEM
//
//  GATEWAY CONFIGURATION

//Addresses
#define UNIT_MAC           0x01  // The address of this gateway

#define ESPNOW_NEIGHBOR_1  0x04  // Address of ESP-NOW neighbor #1
#define ESPNOW_NEIGHBOR_2  0x00  // Address of ESP-NOW neighbor #2
#define LORA_NEIGHBOR_1    0x02  // Address of LoRa neighbor #1
#define LORA_NEIGHBOR_2    0x00  // Address of LoRa neighbor #2

// Interfaces
#define USE_ESPNOW  
#define USE_LORA
//#define USE_WIFI  // Will cause errors if used with ESP-NOW. Use a serial link instead!
//#define USE_ETHERNET

// Routing
// Options: sendESPNowNbr(1 or 2); sendESPNowPeers(); sendLoRaNbr(1 or 2); broadcastLoRa(); sendSerial(); sendMQTT();
#define ESPNOWG_ACT    sendSerial();
#define LORAG_ACT      sendSerial();
#define SERIAL_ACT     sendESPNowNbr(1); sendESPNowPeers(); sendLoRaNbr(1); broadcastLoRa(); //
#define MQTT_ACT          
#define INTERNAL_ACT   sendSerial();
#define ESPNOW1_ACT    sendSerial();
#define ESPNOW2_ACT                   
#define LORA1_ACT      sendSerial();
#define LORA2_ACT`
```
And here is the LoRa repeater config
```
//  FARM DATA RELAY SYSTEM
//
//  GATEWAY CONFIGURATION

//Addresses
#define UNIT_MAC           0x02   // The address of this gateway

#define ESPNOW_NEIGHBOR_1  0x00  // Address of ESP-NOW neighbor #1
#define ESPNOW_NEIGHBOR_2  0x00  // Address of ESP-NOW neighbor #2
#define LORA_NEIGHBOR_1    0x01  // Address of LoRa neighbor #1
#define LORA_NEIGHBOR_2    0x00  // Address of LoRa neighbor #2

// Interfaces
#define USE_ESPNOW  
#define USE_LORA
//#define USE_WIFI  // Will cause errors if used with ESP-NOW. Use a serial link instead!
//#define USE_ETHERNET

// Routing
// Options: sendESPNowNbr(1 or 2); sendESPNowPeers(); sendLoRaNbr(1 or 2); broadcastLoRa(); sendSerial(); sendMQTT();
#define ESPNOWG_ACT    sendLoRaNbr(1);
#define LORAG_ACT      sendLoRaNbr(1);
#define SERIAL_ACT     
#define MQTT_ACT          
#define INTERNAL_ACT   sendLoRaNbr(1);
#define ESPNOW1_ACT    
#define ESPNOW2_ACT                    
#define LORA1_ACT      sendESPNowPeers(); broadcastLoRa(); 
#define LORA2_ACT
```

Thanks for the assist on this, let me know if I have something misconfigured.

**Reply by timmbogner** [reply]
Your configs do look correct. I'm going to test them later today or tonight and then I'll get back to you.

**Comment by aviateur17** [comment]
I don't think the hooks are in the code to register LoRa controllers with the gateways like ESP-NOW does.  I'll see if I can program this up in short manner.

**Reply by timmbogner** [reply]
Not sure what your intention is, but... The intended behavior is that gateways should send broadcasts with 0xFF as the address. Nodes (controllers) receive all broadcasts internally, but only deliver them to the user if the broadcast came from it's own gateway. It would be too radio-heavy to send individual transmissions to each LoRa address as ESP-NOW does.

With that in mind, I can't think of a reason to register individual LoRa devices with gateways. Let me know if you have something in mind!

The configs here look okay. I'm going to test and see what's up later on.

**Reply by veronal** [reply]
Just to be clear the controller in question is a espNOW controller.   It talks to the LoRa repeater via espNow,

**Comment by timmbogner** [comment]
Hey @veronal ,
I apologize for the delay. This slipped by me while I was dealing with other issues. Did you ever get things figured out? I tested it and can't recreate your issue. The following is what your controller's config should look like, and you should make sure you're subscribed to the ID you are sending with too.
```
#define READING_ID    1   //Unique ID for this sensor
#define GTWY_MAC      0x02 //Address of the nearest gateway

#define USE_ESPNOW
//#define USE_LORA
```
